pydm/widgets/waveformplot.py

```diff
@@ -5,85 +5,204 @@
-    # just call redraw if y_waveform has vale
-    if self.y_waveform is not None:
-        self.redrawPlot()
-  
-  @pyqtSlot(np.ndarray)
-  def receiveYWaveform(self, new_waveform):
-    self.y_waveform = new_waveform
-    self.redrawPlot()
-
-  def updateAxes(self):
-    if self.y_waveform is None:
-      return
-    if self.x_waveform is None:
-      yspan = float(np.amax(self.y_waveform)) - float(np.amin(self.y_waveform))
-      self.plotItem.setLimits(xMin=0, xMax=len(self.y_waveform), yMin=float(np.amin(self.y_waveform)-yspan), yMax=float(np.amax(self.y_waveform)+yspan))
-    else:
-      # let the y and x shapes equals
-      if self.x_waveform.shape[0] > self.y_waveform.shape[0]:
-          self.x_waveform = self.x_waveform[:self.y_waveform.shape[0]]
-      elif self.x_waveform.shape[0] < self.y_waveform.shape[0]:
-          self.y_waveform = self.y_waveform[:self.x_waveform.shape[0]]
-      self.plotItem.setLimits(xMin=np.amin(self.x_waveform), xMax=np.amax(self.x_waveform), yMin=np.amin(self.y_waveform), yMax=np.amax(self.y_waveform))
-  
-  def redrawPlot(self):
-    self.updateAxes()
-    self.curve.setData(x=self.x_waveform, y=self.y_waveform)
-  
-  # -2 to +2, -2 is LOLO, -1 is LOW, 0 is OK, etc.  
-  @pyqtSlot(int)
-  def alarmStatusChanged(self, new_alarm_state):
-    pass
-  
-  #0 = NO_ALARM, 1 = MINOR, 2 = MAJOR, 3 = INVALID  
-  @pyqtSlot(int)
-  def alarmSeverityChanged(self, new_alarm_severity):
-    pass
-    
-  #false = disconnected, true = connected
-  @pyqtSlot(bool)
-  def connectionStateChanged(self, connected):
-    pass
-  
-  def getYChannel(self):
-    return str(self._ychannel)
-  
-  def setYChannel(self, value):
-    if self._ychannel != value:
-      self._ychannel = str(value)
+class NoDataError(Exception):
+	pass
 
-  def resetYChannel(self):
-    if self._ychannel != None:
-      self._ychannel = None
-    
-  yChannel = pyqtProperty(str, getYChannel, setYChannel, resetYChannel)
-  
-  def getXChannel(self):
-    return str(self._xchannel)
-  
-  def setXChannel(self, value):
-    if self._xchannel != value:
-      self._xchannel = str(value)
-
-  def resetXChannel(self):
-    if self._xchannel != None:
-      self._xchannel = None
-    
-  xChannel = pyqtProperty(str, getXChannel, setXChannel, resetXChannel)
-  
-  def channels(self):
-    return [PyDMChannel(address=self.yChannel, connection_slot=self.connectionStateChanged, waveform_slot=self.receiveYWaveform, severity_slot=self.alarmSeverityChanged), PyDMChannel(address=self.xChannel, connection_slot=self.connectionStateChanged, waveform_slot=self.receiveXWaveform, severity_slot=self.alarmSeverityChanged)]
+class WaveformCurveItem(PlotCurveItem):
+	data_changed = pyqtSignal()
+	def __init__(self, y_addr=None, x_addr=None, **kws):
+		if y_addr is None:
+			y_addr = ""
+		if 'name' not in kws or kws['name'] is None:
+			try:
+				y_name = y_addr.split("://")[1]
+			except IndexError:
+				y_name = y_addr
+			if x_addr is None:
```

Comment:
Also, it would be very good to have this ```remove_protocol``` in the utils of PyDM since it will be used in more places! ;)